sov-modules/sov-modules-api/src/default_signature.rs

```diff
@@ -1,50 +1,100 @@
+    pub struct DefaultPrivateKey {
+        key_pair: Keypair,
     }
 
-    pub fn sign(&self, _msg: [u8; 32]) -> DefaultSignature {
-        DefaultSignature {
-            msg_sig: vec![],
-            should_fail: false,
+    impl DefaultPrivateKey {
+        pub fn generate() -> Self {
+            let mut csprng = OsRng;
+
+            Self {
+                key_pair: Keypair::generate(&mut csprng),
+            }
+        }
+
+        pub fn sign(&self, msg: [u8; 32]) -> DefaultSignature {
+            DefaultSignature {
+                msg_sig: self.key_pair.sign(&msg),
+            }
+        }
+
+        pub fn pub_key(&self) -> DefaultPublicKey {
+            DefaultPublicKey {
+                pub_key: self.key_pair.public,
+            }
         }
     }
 }
 
-impl<T: AsRef<str>> From<T> for DefaultPublicKey {
-    fn from(key: T) -> Self {
-        let key = key.as_ref().as_bytes().to_vec();
-        Self { pub_key: key }
+#[derive(PartialEq, Eq, Clone, Debug)]
+pub struct DefaultPublicKey {
+    pub(crate) pub_key: DalekPublicKey,
+}
+
+impl BorshDeserialize for DefaultPublicKey {
+    fn deserialize_reader<R: std::io::Read>(reader: &mut R) -> std::io::Result<Self> {
+        let mut buffer = [0; PUBLIC_KEY_LENGTH];
+        reader.read_exact(&mut buffer).unwrap();
+
+        Ok(Self {
+            pub_key: DalekPublicKey::from_bytes(&buffer).unwrap(),
+        })
     }
 }
 
-#[derive(borsh::BorshDeserialize, borsh::BorshSerialize, PartialEq, Eq, Debug, Clone, Default)]
+impl BorshSerialize for DefaultPublicKey {
+    fn serialize<W: std::io::Write>(&self, writer: &mut W) -> std::io::Result<()> {
+        writer.write_all(self.pub_key.as_bytes())
+    }
+}
+
+#[derive(PartialEq, Eq, Debug, Clone)]
 pub struct DefaultSignature {
-    pub msg_sig: Vec<u8>,
-    pub should_fail: bool,
+    pub msg_sig: DalekSignature,
+}
+
+impl BorshDeserialize for DefaultSignature {
+    fn deserialize_reader<R: std::io::Read>(reader: &mut R) -> std::io::Result<Self> {
+        let mut buffer = [0; SIGNATURE_LENGTH];
+        reader.read_exact(&mut buffer).unwrap();
+
+        Ok(Self {
+            msg_sig: DalekSignature::from_bytes(&buffer).unwrap(),
+        })
+    }
+}
+
+impl BorshSerialize for DefaultSignature {
+    fn serialize<W: std::io::Write>(&self, writer: &mut W) -> std::io::Result<()> {
+        writer.write_all(self.msg_sig.as_bytes())
+    }
 }
 
 impl Signature for DefaultSignature {
     type PublicKey = DefaultPublicKey;
 
     fn verify(
         &self,
-        _pub_key: &Self::PublicKey,
-        _msg_hash: [u8; 32],
+        pub_key: &Self::PublicKey,
+        msg_hash: [u8; 32],
     ) -> Result<(), SigVerificationError> {
-        if self.should_fail {
-            Err(SigVerificationError::BadSignature)
-        } else {
-            Ok(())
-        }
+        pub_key
+            .pub_key
+            .verify_strict(&msg_hash, &self.msg_sig)
+            .map_err(|_| SigVerificationError::BadSignature)
```

Comment:
We swallowing exact error, which is not ideal, but we don't have logging anyway. 
Maybe `BadSignature` at least could have string inside?

Reply:
Sorry, I don't see the change.

Reply:
It is there but PR diff is not showing it: https://github.com/Sovereign-Labs/sovereign/blob/blaze/dalek_sig/sov-modules/sov-modules-api/src/default_signature.rs#L100